### PR TITLE
feat: [CDB-226]: Custom Dashboards - Implemented Persistent Filtering

### PR DESCRIPTION
**Pull request by Harness-JKhattak**
##### Summary:

- Implemented Persistent Filtering for the Dashboards page.
- URL Query Parameters are updated to match what filters the user has selected
- Query Params are as follows:
 -- view - grid/list
 -- labels - string array
 -- sort - Name, Recently Viewed, Recently Created, Most Viewed, Most Liked
 -- tags - string array
 -- page - number
 -- search - string

##### Jira Links:

[CDB-226](https://harness.atlassian.net/browse/CDB-226)

##### Video:

https://user-images.githubusercontent.com/103499516/174071519-b10efba6-d41f-41db-9a42-e25e6d81feff.mov

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by Harness-JKhattak** [comment]
retrigger jest

File: src/modules/45-dashboards/types/DashboardQueryParams.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+export enum SortEnum {
+  MOST_LIKED = 'most_liked',
+  MOST_VIEWED = 'most_viewed',
+  NAME = 'title',
+  RECENTLY_CREATED = 'recently_created',
+  RECENTLY_VIEWED = 'recently_viewed'
+}
+
+export interface DashboardQueryParams {
```
Comment (sean-harness) [inline src/modules/45-dashboards/types/DashboardQueryParams.tsx:18]: Is this type being used?

**Comment by Harness-JKhattak** [comment]
trigger cypress

**Review by sean-harness — APPROVED** [review]

**Comment by Harness-JKhattak** [comment]
trigger cypress

**Comment by Harness-JKhattak** [comment]
trigger cypress

**Review by laura-mannus — APPROVED** [review]

**Pull request merged**